hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -74,13 +75,16 @@ public final class HoodieFileGroupReader<T> implements Closeable {
   // Length of bytes to read from the base file
   private final long length;
   // Key to record and metadata mapping from log files
+  private final boolean positionBasedWriteEnabled;
   private final Map<String, Pair<Option<T>, Map<String, Object>>> logFileRecordMapping = new HashMap<>();
+  private final Map<Integer, Pair<Option<T>, Map<String, Object>>> positionToRecords = new HashMap<>();
```

Comment:
i'm wondering if we should use ConcurrentHashMap? There can be concurrent calls to `scanLogFies`, when I want to read two file slices at the same time.

Reply:
Make sense.